subsys/bluetooth/host/classic/rfcomm.c

```diff
@@ -1427,6 +1427,10 @@ static void rfcomm_handle_data(struct bt_rfcomm_session *session,
 	}
 
 	if (pf == BT_RFCOMM_PF_UIH_CREDIT) {
+		if (buf->len == 0) {
+			LOG_ERR("Data recvd is invalid");
```

Comment:
I was unsure about the use of `LOG_ERR`, but the function seems to use that for other errors in this function so it seems OK or alternatively all of them should be changed to e.g. `LOG_DBG`

Reply:
Yep, it didn't seem a fatal error for me. But since other places in this code is using ERR, I just followed this pattern.

Reply:
Generally, the approach that we've tried to take in the Bluetooth stack, is to use `LOG_WRN()` for situations where we receive invalid or unexpected data from a remote device, whereas `LOG_ERR()` is for local failures.

Reply:
As per the documentation, `LOG_ERR` should only be used in unrecoverable states: https://docs.zephyrproject.org/latest/services/logging/index.html#c.LOG_ERR

Reply:
> As per the documentation, `LOG_ERR` should only be used in unrecoverable states: https://docs.zephyrproject.org/latest/services/logging/index.html#c.LOG_ERR

It doesn't quite seem to say that, rather gives it as an example where LOG_ERR() should ne used. Invalid data from a remote device is in practice always recoverable though (ignore it, disconnect the connection, etc).

Reply:
> > As per the documentation, `LOG_ERR` should only be used in unrecoverable states: https://docs.zephyrproject.org/latest/services/logging/index.html#c.LOG_ERR
> 
> It doesn't quite seem to say that, rather gives it as an example where LOG_ERR() should ne used. Invalid data from a remote device is in practice always recoverable though (ignore it, disconnect the connection, etc).

From the documentation:

> It’s meant to report severe errors, such as those from which it’s not possible to recover.

"not possible to recover" == "unrecoverable" in my understanding :) 
But it's true that it does not say that it should not be used for other conditions, but it is certainly intended for those.

Reply:
"such as" means (at least to me) "as an example", i.e. it doesn't exclude other uses.

Reply:
@ceolin could you update this to `LOG_WRN()` so we get this merged? Thanks.